src/libcore/pin.rs

```diff
@@ -584,6 +584,25 @@ impl<P: DerefMut> Pin<P> {
     /// the pointee cannot move after `Pin<Pointer<T>>` got created.
     /// "Malicious" implementations of `Pointer::DerefMut` are likewise
     /// ruled out by the contract of `Pin::new_unchecked`.
+    ///
+    /// This method is useful when doing multiple calls to functions that consume the pinned type.
+    ///
+    /// ```
```

Comment:
I think usually we have `# Example` or so above the example, might be good to have that here, too, for consistency.

Reply:
Done.